src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/DetectorOptions.java

```diff
@@ -41,7 +41,7 @@ public final class DetectorOptions {
      */
     private String format;
     /** The message to report on detection. If blank, then use the format. */
-    private String message = "";
+    private String message;
```

Comment:
This is property of check.
We make it nullable, but user can not set null by config, so user can not bring Check back to default state.
Not good.

Even Javadoc tells us that empty is special value. We need to change javadoc.

But really we need to change code. Can remove `Optional.ofNullable(message).orElse("")` and treat message as not nullable?

Reply:
I got your point. but the check which use this builder Pattern such like https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java 

In this check itself docs says that by default value is null so even if user can't set null but still by default value is null and everytime https://github.com/checkstyle/checkstyle/blob/58fdccf566a46b2ad4eea3987be3157d323bb3a8/src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java#L218-L227 
we call the build method to get the value 
https://github.com/checkstyle/checkstyle/blob/58fdccf566a46b2ad4eea3987be3157d323bb3a8/src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpSinglelineCheck.java#L206
so when user doesn't set any value then message become null so we can't remove ` Optional.ofNullable(message).orElse("")`. 

and I also https://checkstyle.org/property_types.html don't see this in docs

Reply:
This is my bad.
It is not a Module! 
It is some inner class, and all fields already not initialized, so with such update we do implementation consistent.